lib/ts/jemallctl.cc

```diff
@@ -0,0 +1,261 @@
+/** @file
+
+  Memory allocation routines for libts
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "ts/jemallctl.h"
+
+#include "ts/ink_platform.h"
+
+// includes jemalloc.h
+#include "ts/ink_memory.h"
+#include "ts/ink_defs.h"
+#include "ts/ink_stack_trace.h"
+#include "ts/Diags.h"
+#include "ts/ink_atomic.h"
+#include "ts/ink_align.h"
+
+#include <cassert>
+#if defined(linux) && !defined(_XOPEN_SOURCE)
+#define _XOPEN_SOURCE 600
```

Comment:
Previously to what? Brought in from where? This entire file appears to be new, and I'm not seeing a corresponding removal that would suggest that it was moved.